lib/env.h

```diff
@@ -43,12 +43,15 @@
 #endif
 
 // Detect word size:
-#if defined(_INTEGRAL_MAX_BITS)
+#if defined(__x86_64__) && defined(__ILP32__)
+// this is the way to detect x32 ABI per https://wiki.debian.org/X32Port
+#  define BASE64_WORDSIZE 64
```

Comment:
I wanted to keep the old behavior for platforms other than x32 ABI but this check could be simplified in this specific instance to just:
```
#if defined(__x86_64__)
  // force word size on __x86_64__ 
  // on x32 ABI which defines __x86_64__ and uses 64-bit words, __SIZE_WIDTH__ == 32
  #  define BASE64_WORDSIZE 64
```

Reply:
Indeed, I'll fix it in the rebase.